Add troubleshooting guide for corrupt repository (#88391)
elastic/elasticsearch: docs/reference/tab-widgets/troubleshooting/snapshot/corrupt-repository.asciidoc

right side of the repository to be marked as read-only. On the Edit page that opened 
scroll down and check "Read-only repository". Click "Save".
Alternatively if deleting the repository altogether is prefered select the checkbox
at the left of the repository name in the repositories table and click the 
"Remove repository" red button at the top left of the table.

At this point, it's only the primary (current) deployment that has the repository marked
as writeable.
{es} sees it as corrupt, so the repository needs to be removed and added back so that 
{es} can resume using it:

Note that we're now configuring the primary (current) deployment.

. Open the primary deployment's side navigation menu (placed under the Elastic logo in the upper left corner)
and go to **Stack Management > Snapshot and Restore > Repositories**.
+
[role="screenshot"]
image::images/repositories.png[{kib} Console,align="center"]

. Get the details for the repository we'll recreate later by clicking on the repository
name in the repositories table and making note of all the repository configurations
that are displayed on the repository details page (we'll use them when we recreate

> the repository). Close the details page using the "X Close" link at
> the bottom left of the page.

Reviewer: I think taking out the text will help in the future if it ever changes (totally optional though)

```suggestion
the repository). Close the details page using the link at
the bottom left of the page.
```

+
[role="screenshot"]
image::images/repo_details.png[{kib} Console,align="center"]

> . With all the details above noted down, let's delete the repository. Select the 

Reviewer: ```suggestion
. With all the details above noted, next delete the repository. Select the 
```

and go to **Stack Management > Snapshot and Restore > Repositories**.
+
[role="screenshot"]
image::images/repositories.png[{kib} Console,align="center"]

. The repositories table should now be visible. Click on the pencil icon at the 
right side of the repository to be marked as read-only. On the Edit page that opened 
scroll down and check "Read-only repository". Click "Save".
Alternatively if deleting the repository altogether is prefered select the checkbox
at the left of the repository name in the repositories table and click the 
"Remove repository" red button at the top left of the table.

At this point, it's only the main (current) deployment that has the repository marked
as writeable.
{es} sees it as corrupt though so let's remove the repository and re-add it so that 
{es} can resume using it:

Note that now we're configuring the main (current) deployment.

. Open the main deployment's side navigation menu (placed under the Elastic logo in the upper left corner)
and go to **Stack Management > Snapshot and Restore > Repositories**.
+
[role="screenshot"]
image::images/repositories.png[{kib} Console,align="center"]

> . Get the details for the repository we'll readd by clicking on the repository

Reviewer: ```suggestion
. Get the details for the repository to re-add by clicking on the repository
```

image::images/repo_details.png[{kib} Console,align="center"]

. With all the details above noted down, let's delete the repository. Select the 
checkbox at the left of the repository and hit the "Remove repository" red button
at the top left of the page.

. Re-add the repository to the system by clicking the "Register Repository" button
at the top right corner of the repositories table. 
+
[role="screenshot"]
image::images/register_repo.png[{kib} Console,align="center"]

. Fill in the repository name, select the type and click "Next".
+
[role="screenshot"]
image::images/register_repo_details.png[{kib} Console,align="center"]

. Fill in the repository details (client, bucket, base path etc) with the values
you noted down before deleting the repository and click the "Register" button
at the bottom.

. Select "Verify repository" to confirm that your settings are correct and the 
deployment can connect to your repository. 
//end::kibana-api-ex[]
// end::cloud[]

// tag::self-managed[]
Fixing the corrupted repository will entail making changes in multiple clusters
that write to the same snapshot repository.
Only one cluster must be writing to a repository. Let's call the cluster 

> we want to keep writing to the repository the "main" cluster (the current cluster), 

Reviewer: ```suggestion
we want to keep writing to the repository the "primary" cluster (the current cluster), 
```

and the other one(s) where we'll mark the repository as read-only the "secondary" 
clusters.

Let's first work on the secondary clusters:

. Get the configuration of the repository:
+
[source,console]
----
GET _snapshot/my-repo
----
// TEST[skip:we're not setting up repos in these tests]
+
The reponse will look like this:
+
[source,console-result]
----
{
  "my-repo": { <1>
    "type": "s3",
    "settings": {
      "bucket": "repo-bucket",
      "client": "elastic-internal-71bcd3",
      "base_path": "myrepo"
    }
  }
}
----
// TESTRESPONSE[skip:the result is for illustrating purposes only]
+
<1> Represents the current configuration for the repository.

> . Using the settings retrieve above let's add the `readonly: true` option to mark
> it as read-only:

Reviewer: ```suggestion
. Using the settings retrieved above, add the `readonly: true` option to mark
it as read-only:
```

+
[source,console]
----
PUT _snapshot/my-repo
{
    "type": "s3",
    "settings": {
      "bucket": "repo-bucket",
      "client": "elastic-internal-71bcd3",
      "base_path": "myrepo",
      "readonly": true <1>
    }
}
----
// TEST[skip:we're not setting up repos in these tests]
+
<1> Marks the repository as read-only.

. Alternatively, deleting the repository is an option using:
+
[source,console]
----
DELETE _snapshot/my-repo
----
// TEST[skip:we're not setting up repos in these tests]
+
The response will look like this:
+
[source,console-result]
------------------------------------------------------------------------------
{
  "acknowledged": true
}
------------------------------------------------------------------------------
// TESTRESPONSE[skip:the result is for illustrating purposes only]

> At this point, it's only the main (current) cluster that has the repository marked

Reviewer: ```suggestion
At this point, it's only the primary (current) cluster that has the repository marked
```

as writeable.
{es} sees it as corrupt though so let's remove the repository and re-add it so that 
{es} can resume using it:

> Note that now we're configuring the main (current) cluster.

Reviewer: ```suggestion
Note that now we're configuring the primary (current) cluster.
```
